lib/tools.js

```diff
@@ -597,6 +598,88 @@ function buildCertChain(cert, ca) {
         .join('\n');
 }
 
+function checkPwnedPassword(password, cache) {
+    const PREFIX_LENGTH = 5;
+    const API_URL = 'https://api.pwnedpasswords.com/range/';
+    const API_TIMEOUT = 1000;
+    const HTTP_STATUS_OK = 200;
+    const HTTP_STATUS_NOT_FOUND = 404;
+
+    function hash(password) {
+        const shasum = crypto.createHash('sha1');
+        shasum.update(password);
+        return shasum.digest('hex');
+    }
+
+    function get(hashedPasswordPrefix) {
+        const opts = {
+            timeout: API_TIMEOUT
+        };
+
+        return new Promise((resolve, reject) => {
+            const req = https
```

Comment:
Why not use the already included pwndpasswords library? https://github.com/zone-eu/wildduck/blob/d8577623b9fc7749d511ad5df6d8ecd4bc3b4e26/lib/api/users.js#L14